plugins/modules/zabbix_hostmacro.py

```diff
@@ -123,24 +134,36 @@ def get_host_macro(self, macro_name, host_id):
             self._module.fail_json(msg="Failed to get host macro %s: %s" % (macro_name, e))
 
     # create host macro
-    def create_host_macro(self, macro_name, macro_value, host_id):
+    def create_host_macro(self, macro_name, macro_value, macro_type, host_id):
         try:
             if self._module.check_mode:
                 self._module.exit_json(changed=True)
-            self._zapi.usermacro.create({'hostid': host_id, 'macro': macro_name, 'value': macro_value})
+            if LooseVersion(self._zbx_api_version) >= LooseVersion('5.0'):
+                self._zapi.usermacro.create({'hostid': host_id, 'macro': macro_name, 'value': macro_value, 'type': macro_type})
+            else:
+                self._zapi.usermacro.create({'hostid': host_id, 'macro': macro_name, 'value': macro_value})
             self._module.exit_json(changed=True, result="Successfully added host macro %s" % macro_name)
         except Exception as e:
             self._module.fail_json(msg="Failed to create host macro %s: %s" % (macro_name, e))
 
     # update host macro
-    def update_host_macro(self, host_macro_obj, macro_name, macro_value):
+    def update_host_macro(self, host_macro_obj, macro_name, macro_value, macro_type):
         host_macro_id = host_macro_obj['hostmacroid']
-        if host_macro_obj['macro'] == macro_name and host_macro_obj['value'] == macro_value:
-            self._module.exit_json(changed=False, result="Host macro %s already up to date" % macro_name)
+        if host_macro_obj['macro'] == macro_name:
+            if LooseVersion(self._zbx_api_version) >= LooseVersion('5.0'):
+                # no change only when macro type == 0. when type = 1 or 2 zabbix will not output value of it.
+                if host_macro_obj['type'] == 0 and macro_type == 0 and host_macro_obj['value'] == macro_value:
```

Comment:
It seems CI failing indicate "wrong decetion of changing".
I guess type of host_macro_obj['type'] is string, so this condition will be false always.

```
TASK [test_zabbix_hostmacro : assert that nothing has been changed] ************
fatal: [testhost]: FAILED! => {
    "assertion": "not zbxhmacro_existing is changed",
    "changed": false,
    "evaluated_to": false,
    "msg": "Assertion failed"
}
```

Reply:
@miettal Thank you for help me pointing out. I've been quite busy with my work recently and should be able to submit a fix in a few days.

Reply:
There is also something wrong with integration testing with Zbx 5.0+:
https://github.com/ansible-collections/community.zabbix/pull/456/checks?check_run_id=3590366728#step:7:2649

Seems like module is not idempotent once macro `type` stirs into the mix

Reply:
You can even test locally if you move you collection to ~/ansible_collections/community/zabbix and then:
```bash
export zabbix_version=5.0
docker-compose up -d
ansible-test integration -v test_zabbix_hostmacro
docker-compose down
```
which is waaay quicker then relying on Github Actions.